api/client/create.go

```diff
@@ -77,7 +85,12 @@ func newCIDFile(path string) (*cidFile, error) {
 	return &cidFile{path: path, file: f}, nil
 }
 
-func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runconfig.HostConfig, cidfile, name string) (*types.ContainerCreateResponse, error) {
+func (cli *DockerCli) createContainer(createConfig *createConfig) (*types.ContainerCreateResponse, error) {
+	config := createConfig.config
+	hostConfig := createConfig.hostConfig
+	cidfile := createConfig.cidfile
+	name := createConfig.name
+	pull := createConfig.pull
```

Comment:
@vdemeester  I just want to avoid a lot of diff and it's much more easy for maintainers to review not use them `inline`-:)